src/Workspace/index.jsx

```diff
@@ -73,19 +74,7 @@ class Workspace extends React.Component {
   }
 
   componentDidMount() {
-    fetchWithCreds({
-      path: `${wtsPath}connected`,
-      method: 'GET',
-    })
-      .then(
-        ({ status }) => {
-          if (status !== 200) {
-            window.location.href = `${wtsPath}/authorization_url?redirect=${window.location.pathname}`;
-          } else {
-            this.connected();
-          }
-        },
-      );
+    initWorkspaceRefreshToken(this.connected);
```

Comment:
so, if this new feature is enabled, when user navigates from a page to workspace, another wts call will be dispatched? seems like a bit redundant

Reply:
Good point... But I think that if we want to do this with a feature flag that is `false` by default, then we will need to keep some of this redundancy for backwards compatibility purposes. Also, an extra refresh is not so bad...it is better than _no_ refresh (which happens if users use the portal but don't go into the workspace for a long period - a scenario we have in our project).

Reply:
Please note that this diff is technically not really a code/behaviour change: I basically moved the code on the left into a generic function that can be used in different places.

Reply:
Another option is to make a breaking change and make a feature flag for workspace as a whole, and make this flag `true` by default. This would simplify the code, but would require some special attention from teams that have a deployment that does not use workspace/ WTS.

Reply:
why not check if `workspaceTokenServiceRefreshTokenAtLogin` is enabled, and only make the call if it's not?

Reply:
That could work, but IMO it is an unnecessary optimisation. But if you prefer this, I can easily add it. Let me know.

Reply:
it makes sense to me... depending on the flag, the call is made either globally, or when the workspace page loads, and not both. But we can defer to @mfshao who brought this up in the first place

Reply:
That is the point I bring this up: we should be able to add a check so this didn't get called repeatly

Reply:
Fixed here: https://github.com/uc-cdis/data-portal/pull/1095/commits/882043f31b3f99be0e18cc94337e64b8ba8ae442